controllers/articles.controllers.js

```diff
@@ -24,4 +26,17 @@ exports.getArticleCommentsById = (req, res, next) => {
         res.status(200).send({comments:resolvedPromises[0]})
     })
     .catch(next)
+}
+exports.postArticleCommentById = (req, res, next) => {
+    const article_id=req.params.article_id
+    const {username,body}=req.body
+
+    Promise.all([checkArticleExists(article_id),checkUsersExists(username)])
```

Comment:
Beautiful function! However, article_id and author are foreign keys, so PSQL will throw an error (that you can handle with a [PSQL error handler](https://notes.northcoders.com/courses/js-back-end/error-handling#psql-thrown-errors)) if you try and insert a non-existent or invalid value. 

If you were to take these checks out, i'd imagine that your PSQL error handlers - provided they're set up correctly - should still catch these.